### PR TITLE
[BD-46] feat: refactoring design tokens folders structure

### LINKED ISSUE #2017: Refactor `tokens` file/folder structure to support additional theme variants
# Description

The current structure of the `tokens` directory contains a top-level `src` folder containing all the tokens for the design system (e.g., `global`, `alias`, `component`). As described in https://github.com/openedx/paragon/issues/2014, these tokens are largely a combination of core tokens and tokens representing the values for the default "light" theme variant.

To be forward thinking to support a future dark mode (i.e., an additional theme variant), let's extract the core tokens out from the theme-specific tokens. I might recommend a folder/file structure similar to:

```shell
tokens/src/core/
tokens/src/themes/light/
tokens/src/themes/dark/ # in the future
```

By accomplishing this issue, consuming applications should be able to load the core Paragon styles independently from any theme-variant CSS variables. For example:

```html
<link rel="stylesheet" href="/dark.css" media="(prefers-color-scheme: dark)" />
<link
  rel="stylesheet"
  href="/light.css"
  media="(prefers-color-scheme: light)"
/>
<link rel="stylesheet" href="/core.css" />
```

# Acceptance Criteria
- [x] Ensure the global, alias, and component tokens defined in `./tokens/src/core/` are not specific to a particular theme variant (i.e., light or dark).
- [x] Ensure the global, alias, and component tokens defined in `./tokens/src/themes/light/` are only related to style properties relevant to a light theme variant (most likely color tokens).

### PR DESCRIPTION
**Pull request by PKulkoRaccoonGang**
## Description

- Ensure the global, alias, and component tokens defined in ./tokens/src/core/ are not specific to a particular theme variant (i.e., light or dark).
 - Ensure the global, alias, and component tokens defined in ./tokens/src/themes/light/ are only related to style properties relevant to a light theme variant (most likely color tokens).

Issue: https://github.com/openedx/paragon/issues/2017

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @PKulkoRaccoonGang!


When this pull request is ready, tag your edX technical lead.

File: tokens/sass-helpers.js
Comment (adamstankiewicz) [inline tokens/sass-helpers.js:17]: [curious] Is everything in the `other.json` file specific to the light theme variant? Or are some of the properties, possibly like `theme-interval`, and `yiq-contrasted-threshold`.

Comment (adamstankiewicz) [inline tokens/sass-helpers.js:17]: [question] If/when we have a dark theme variant, I wonder if there is a way we might be able to future proof this `colorYiq` helper function to support multiple theme variants later on (e.g., pulling from `themes/dark/global/other.json` to generate a `dark.css` output)?

The approach we're opting to take here is to have a `light.css` and `dark.css` output file, which would be generated by running the `StyleDictionary.buildAllPlatforms()` twice: once with the light theme variant tokens, and again with the dark theme variant tokens. If there's a way to pass in the current theme variant being used into the `colorYiq` that might be one approach to take here. Open to other ideas of course!

File: tokens/src/themes/light/components/Annotation.json
Comment (adamstankiewicz) [inline tokens/src/themes/light/components/Annotation.json:4]: It looks like the `type` attribute in these tokens might have gotten lost somehow (in the original, now-deleted Annotation.json, the `type` attributes are there. I wonder if it _may_ be related to the issues we ran into trying to get `alpha` brought in sync with `master` that caused us to essentially rewrite the git history for the `alpha` branch... It may be worth a check to see if there might be anything else missing in addition to the `type`? It does look like this may be an issue in many of the files.

**Review by adamstankiewicz — COMMENTED** [review]
Looking good! I left a few comments; biggest feedback is that it looks like the new token files may be missing `type` attributes (and possibly other things?) that were in the original tokens.

File: tokens/sass-helpers.js
Comment (viktorrusakov) [reply tokens/sass-helpers.js]: yes, `theme-interval` and `yiq-contrasted-threshold` are not theme-specific, moved the to the `core` tokens

Comment (viktorrusakov) [reply tokens/sass-helpers.js]: Makes sense, I've modified our functions to check whether token belongs to any theme and take theme's colors (defaults to `light`)

File: tokens/sass-helpers.js
Comment (adamstankiewicz) [inline tokens/sass-helpers.js:14]: nit: `./src/themes/{themeVariant}/global/other.json`

Comment (adamstankiewicz) [inline tokens/sass-helpers.js:14]: [clarification] I don't necessarily think the theme variants are necessarily defaulting to `yiq-text-light` or `yiq-text-dark` depending on the theme variant itself. That choice depends on the `color` being modified (yiq-ified? 😅).

Should the JSDoc comment here just reference that the `light` theme pulls both yiq light/dark values from the `other.json` file in `light` or `dark` folder, respectively, without an explicit mention of `yiq-text-light` or `yiq-text-dark` here?

File: tokens/style-dictionary.js
Comment (viktorrusakov) [reply tokens/style-dictionary.js:34]: actually, nevermind... I just saw that you could define transforms directly in the `config` object without using `registerTransform` method, so we could just use that

Comment (adamstankiewicz) [reply tokens/style-dictionary.js:34]: Ah, yeah, that should theoretically also work! I defer to you if we want to roll with what you have now in this PR to get this PR merged and treat that refactor as a fast-follow or if we want to try to get that change into this PR before merge.

Comment (adamstankiewicz) [reply tokens/style-dictionary.js:34]: We will tackle this as a fast follow in https://github.com/openedx/paragon/issues/2014

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@PKulkoRaccoonGang 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/2069&entry.1484655318=https://github.com/PKulkoRaccoonGang&entry.752974735=2023-02-21+22:51&entry.1917517419=2023-02-24+14:57&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
